release-note-script/src/main/java/ReleaseNoteCreation.java

```diff
@@ -0,0 +1,331 @@
+                }
+              });
+    }
+  }
+
+  private void merge(ReleaseNoteText from, ReleaseNoteText to) {
+    if (from.text != null && !from.text.isEmpty()) {
+      to.text = to.text + " " + from.text;
+    }
+    if (DEBUG != null) System.err.printf("merged RN text: %s%n", to.text);
+    to.prNumbers.addAll(from.prNumbers);
+  }
+
+  void outputReleaseNote() {
+    StringBuilder builder = new StringBuilder();
+    builder.append("## Summary\n\n");
+
+    Arrays.stream(Category.values())
+        .forEach(
+            category -> {
+              List<ReleaseNoteText> releaseNotes = categoryMap.get(category);
+              if (releaseNotes != null && !releaseNotes.isEmpty()) {
+                builder.append(String.format("## %s%n", category.getDisplayName()));
+                builder.append(getFormattedReleaseNotes(releaseNotes)).append("\n");
+              }
+            });
+    System.out.println(builder);
+  }
+
+  private String getFormattedReleaseNotes(List<ReleaseNoteText> releaseNotes) {
+    StringBuilder builder = new StringBuilder();
+    releaseNotes.forEach(
+        rnText -> {
+          builder.append(String.format("- %s (", rnText.text));
+          rnText.prNumbers.forEach(prNum -> builder.append(String.format("#%s ", prNum)));
+          builder.deleteCharAt(builder.length() - 1); // delete the last space character
+          builder.append(")\n");
+        });
+    return builder.toString();
+  }
+
+  private void checkReleaseNoteCategory(ReleaseNoteText rnText) {
+    if (rnText.category == null) {
+      rnText.category = Category.MISCELLANEOUS;
+    }
+  }
+
+  private boolean isContainedInSameAsItems(ReleaseNoteText rnText) {
+    return sameAsItems.values().stream().anyMatch(items -> items.contains(rnText));
+  }
+
+  public enum Category {
+    ENHANCEMENT("Enhancements"),
+    IMPROVEMENT("Improvements"),
+    BUGFIX("Bug fixes"),
+    DOCUMENTATION("Documentation"),
+    MISCELLANEOUS("Miscellaneous");
+
+    private final String displayName;
+
+    Category(String displayName) {
+      this.displayName = displayName;
+    }
+
+    public String getDisplayName() {
+      return this.displayName;
+    }
+
+    public static Category getByName(String name) {
+      return Arrays.stream(Category.values())
+          .filter(v -> v.name().equalsIgnoreCase(name))
+          .findFirst()
+          .orElseThrow(() -> new IllegalArgumentException("Invalid name: " + name));
+    }
+  }
+
+  static class ReleaseNoteText {
+    public Category category;
+    public String text;
+    public List<String> prNumbers = new ArrayList<>();
+  }
+
+  public static class GitHubContext {
+
+    private static final String MERGED_STATE = "merged";
+
+    private final String owner;
+    private final String projectTitlePrefix;
+    private final String version;
+    private final String repository;
+
+    public GitHubContext(String owner, String projectTitString, String version, String repository) {
```

Comment:
I would recommend separating methods to fetch GitHub information (currently via subprocess) and methods to process that information.

Reply:
I see. I'll revise it. Thank you.

Reply:
will do another PR in the future.